config.go

```diff
@@ -30,12 +30,51 @@ import (
 type config struct {
 	Global struct {
 	}
-	Modules map[string]*moduleConfig
-	XXX     map[string]interface{} `yaml:",inline"`
+	Modules   map[string]*moduleConfig
+	Discovery *discoveryConfig
+	XXX       map[string]interface{} `yaml:",inline"`
 
 	bearerToken   string
 	proxyPath     string
 	telemetryPath string
+
+	mutex sync.Mutex
```

Comment:
This should be a reference else mutex locking doesnt work that great

Reply:
It's normal not to use a pointer here, as long as the config object is only ever treated as a pointer. vet is good at warning if the struct gets copied in this case.